crates/librqbit/src/torrent_state/live/mod.rs

```diff
@@ -1018,17 +1020,27 @@ impl PeerHandler {
 
         pe.value_mut().state.set(PeerState::Dead, &pstats);
 
+        if self.incoming {
```

Comment:
please put a debug!() line here

Reply:
Looking at the surprising test failure - you might need to call self.state.peers.drop_peer(handle) before returning. Not sure, please take a look.

Reply:
Test now works, though I do not understand the reason for failure - when only incoming peers were retried - I just then made some changes to logging and now cannot recreate test failure.